# Import Libraries

In [15]:
import pandas as pd
import numpy as np
import os

# 01 Importing Data

# importing order.csv

In [16]:
df= pd.read_csv( r'C:\Users\João Pedro\MAR-Instacart Basket Analysis\02 Data\Original Data\orders.csv', index_col = False)

# creating folder path

In [17]:
path = r'C:\Users\João Pedro\MAR-Instacart Basket Analysis'

In [18]:
path

'C:\\Users\\João Pedro\\MAR-Instacart Basket Analysis'

# Creating vars list without the eval_set column


In [19]:
vars_list = ( 'order_id' , 'user_id' , 'order_number' ,'order_dow' ,'order_hour_of_day' , 'days_since_prior_order')

In [22]:
df = pd.read_csv(os.path.join(path,'02 Data','Original Data','orders.csv'), usecols = vars_list)

# printing first 5 rows of dataframe

In [23]:
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

# Importing products.csv

In [25]:
df_prods = pd.read_csv(os.path.join(path, '02 Data','Original Data','products.csv'))

# Print first 20 rows and the last 35 rows of df_prods dataframe

In [26]:
df_prods.head(20) , df_prods.tail(35)

(    product_id                                       product_name  aisle_id  \
 0            1                         Chocolate Sandwich Cookies        61   
 1            2                                   All-Seasons Salt       104   
 2            3               Robust Golden Unsweetened Oolong Tea        94   
 3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
 4            5                          Green Chile Anytime Sauce         5   
 5            6                                       Dry Nose Oil        11   
 6            7                     Pure Coconut Water With Orange        98   
 7            8                  Cut Russet Potatoes Steam N' Mash       116   
 8            9                  Light Strawberry Blueberry Yogurt       120   
 9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
 10          11                                  Peach Mango Juice        31   
 11          12                         

# Print the names of columns in df_prods dataframe

In [27]:
df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

# Print the number of rows and columns in df_prods dataframe

In [28]:
df_prods.shape

(49693, 5)

In [30]:
df_prods.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

In [32]:
df_prods.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49693 entries, 0 to 49692
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   product_id     49693 non-null  int64  
 1   product_name   49677 non-null  object 
 2   aisle_id       49693 non-null  int64  
 3   department_id  49693 non-null  int64  
 4   prices         49693 non-null  float64
dtypes: float64(1), int64(3), object(1)
memory usage: 1.9+ MB


# Whats the max value of aisle_id column?

The max value is 134

# Whats the data type of department id ?

In [ ]:
INT64 